assets/scripts/twofactor_submit.js

```diff
@@ -0,0 +1,122 @@
+    if (error) {
+      error = '' + error
+    } else {
+      error = 'The Cozy server is unavailable. Do you have network?'
+    }
+
+    if (!errorPanel) {
+      errorPanel = document.createElement('p')
+      errorPanel.classList.add('wizard-errors', 'u-error')
+      loginField.insertBefore(errorPanel, loginField.firstChild)
+    }
+
+    errorPanel.textContent = error
+    submitButton.removeAttribute('disabled')
+  }
+
+  const onSubmitTwoFactorCode = function(event) {
+    event.preventDefault()
+    submitButton.setAttribute('disabled', true)
+
+    const longRunSession =
+      longRunSessionCheckbox && longRunSessionCheckbox.checked ? '1' : '0'
+    const passcode = twoFactorPasscodeInput.value
+    const token = twoFactorTokenInput.value
+    const trustDevice = twoFactorTrustDeviceCheckbox.checked ? '1' : '0'
+    const redirect = redirectInput.value + window.location.hash
+
+    const headers = new Headers()
+    headers.append('Content-Type', 'application/x-www-form-urlencoded')
+    headers.append('Accept', 'application/json')
+
+    const reqBody =
+      'two-factor-passcode=' +
+      encodeURIComponent(passcode) +
+      '&long-run-session=' +
+      encodeURIComponent(longRunSession) +
+      '&two-factor-token=' +
+      encodeURIComponent(token) +
+      '&two-factor-generate-trusted-device-token=' +
+      encodeURIComponent(trustDevice) +
+      '&redirect=' +
+      encodeURIComponent(redirect) +
+      '&csrf_token=' +
+      encodeURIComponent(csrfTokenInput.value)
+    fetch('/auth/twofactor', {
+      method: 'POST',
+      headers: headers,
+      body: reqBody,
+      credentials: 'same-origin'
+    })
+      .then(function(response) {
+        const loginSuccess = response.status < 400
+
+        response
+          .json()
+          .then(function(body) {
+            if (loginSuccess) {
+              if (
+                localStorage &&
+                typeof body.two_factor_trusted_device_token == 'string'
+              ) {
+                localStorage.setItem(
+                  twoFactorTrustedDeviceTokenKey,
+                  body.two_factor_trusted_device_token
+                )
+              }
+              if (body.redirect) {
+                window.location = body.redirect
+              }
+            } else {
+              showError(body.error)
+              twoFactorPasscodeInput.classList.add('is-error')
+              twoFactorPasscodeInput.select()
+            }
+          })
+          .catch(showError)
+      })
+      .catch(showError)
+  }
+
+  // Responsive design
+  if (document.body.clientWidth > 1024) {
+    const avatars = document.getElementsByClassName('c-avatar')
+    for (const avatar of avatars) {
+      avatar.classList.add('c-avatar--xlarge', 'u-mv-1')
+    }
+    const footers = document.getElementsByClassName('wizard-footer')
+    for (const footer of footers) {
+      footer.classList.remove('u-pb-half')
+      footer.classList.add('u-pb-2')
+    }
+  }
```

Comment:
I'd prefer if we can avoid duplicating those lines. Did you ask @GoOz if some responsive utilities CSS classes have been added to cozy-ui that would help to remove this hack?

Reply:
Indeed there is for utilities.

for the avatar I would put `class="u-mv-0-m u-mv-1"`
- `u-mv-1` adds a vertical margin of 1 in any circumstances
- `u-mv-0-m` removes vertical margin for viewports with a width below 1024px

for the footer I would put `class="u-pb-half-m u-pb-2"`
- `u-pb-2` adds a padding-bottom of 2 in any circumstances
- `u-pb-half-m` adds a padding-bottom of half a unit for viewports with a width below 1024px